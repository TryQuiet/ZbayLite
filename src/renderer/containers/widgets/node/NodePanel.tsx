import React from 'react'
import { connect } from 'react-redux'
import * as R from 'ramda'

import NodePanelComponent from '../../../components/widgets/node/NodePanel'
// import nodeSelectors from '../../../store/selectors/node'
// import identitySelectors from '../../../store/selectors/identity'

export const mapStateToProps = _state => ({
  // status: nodeSelectors.status(state),
  // connections: nodeSelectors.connections(state).toNumber(),
  // freeUtxos: identitySelectors.freeUtxos(state)
})

export const NodePanel = ({ status, freeUtxos, connections }) => {
  const [expanded, setExpanded] = React.useState(false)
  return (
    <NodePanelComponent
      expanded={expanded}
      status={status}
      setExpanded={setExpanded}
      freeUtxos={freeUtxos}
      connections={connections}
    />
  )
}

export default R.compose(React.memo, connect(mapStateToProps))(NodePanel)
