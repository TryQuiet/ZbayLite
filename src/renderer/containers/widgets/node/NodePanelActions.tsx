
// import { bindActionCreators } from 'redux'

import NodePanelActions from '../../../components/widgets/node/NodePanelActions'

// import nodeHandlers from '../../../store/handlers/node'

export default NodePanelActions
