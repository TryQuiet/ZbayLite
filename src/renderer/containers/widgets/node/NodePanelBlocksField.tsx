
import React from 'react'
import BigNumber from 'bignumber.js'
import PropTypes from 'prop-types'
import { connect } from 'react-redux'

import NodePanelField from '../../../components/widgets/node/NodePanelField'

// import nodeSelectors from '../../../store/selectors/node'

export const mapStateToProps = _state => ({
  // latestBlock: nodeSelectors.latestBlock(state),
  // currentBlock: nodeSelectors.currentBlock(state)
})

export const NodePanelBlocksField = ({ latestBlock, currentBlock }) => {
  const outOf = latestBlock.isZero() ? '?' : `~${latestBlock.toString()}`
  return <NodePanelField name='Blocks' value={`${currentBlock.toString()} / ${outOf}`} />
}

NodePanelBlocksField.propTypes = {
  latestBlock: PropTypes.instanceOf(BigNumber),
  currentBlock: PropTypes.instanceOf(BigNumber)
}

export default connect(mapStateToProps)(NodePanelBlocksField)
