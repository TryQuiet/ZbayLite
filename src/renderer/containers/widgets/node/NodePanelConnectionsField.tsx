import React from 'react'
import BigNumber from 'bignumber.js'
import PropTypes from 'prop-types'
import { connect } from 'react-redux'

import NodePanelField from '../../../components/widgets/node/NodePanelField'

export const mapStateToProps = _state => ({
  // connections: nodeSelectors.connections(state)
})

export const NodePanelConnectionsField = ({ connections }) => (
  <NodePanelField name='Connections' value={connections.toString()} />
)

NodePanelConnectionsField.propTypes = {
  connections: PropTypes.instanceOf(BigNumber).isRequired
}

export default connect(mapStateToProps)(NodePanelConnectionsField)
