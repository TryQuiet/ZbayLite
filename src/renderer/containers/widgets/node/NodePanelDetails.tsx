import { connect } from 'react-redux'

import NodePanelDetails from '../../../components/widgets/node/NodePanelDetails'

import appSelectors from '../../../store/selectors/app'

export const mapStateToProps = state => ({
  zbayVersion: appSelectors.version(state)
})

export default connect(mapStateToProps)(NodePanelDetails)
