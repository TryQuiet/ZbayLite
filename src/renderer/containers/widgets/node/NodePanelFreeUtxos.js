import React from 'react'
import PropTypes from 'prop-types'
import { connect } from 'react-redux'

import NodePanelField from '../../../components/widgets/node/NodePanelField'

import identitySelectors from '../../../store/selectors/identity'

export const mapStateToProps = state => ({
  freeUtxos: identitySelectors.freeUtxos(state)
})

export const NodePanelFreeUtxos = ({ freeUtxos }) => (
  <NodePanelField name={'FREE UTXO\'S'} value={freeUtxos.toString()} />
)

NodePanelFreeUtxos.propTypes = {
  freeUtxos: PropTypes.number.isRequired
}

export default connect(mapStateToProps)(NodePanelFreeUtxos)
