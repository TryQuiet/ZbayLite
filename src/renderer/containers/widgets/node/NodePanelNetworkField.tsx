
import React from 'react'
import PropTypes from 'prop-types'
import { connect } from 'react-redux'

import NodePanelField from '../../../components/widgets/node/NodePanelField'

// import nodeSelectors from '../../../store/selectors/node'

export const mapStateToProps = _state => ({
  // network: nodeSelectors.network(state)
})

export const NodePanelNetworkField = ({ network }) => (
  <NodePanelField name='Network' value={network} />
)

NodePanelNetworkField.propTypes = {
  network: PropTypes.oneOf(['mainnet', 'testnet'])
}

export default connect(mapStateToProps)(NodePanelNetworkField)
