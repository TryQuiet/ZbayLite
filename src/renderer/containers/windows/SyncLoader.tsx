import { connect } from 'react-redux'
import { bindActionCreators } from 'redux'

// import nodeHandlers from '../../store/handlers/node'
// import nodeSelectors from '../../store/selectors/node'
import SyncLoader from '../../components/windows/SyncLoader'

export const mapStateToProps = _state => ({
  // nodeStatus: nodeSelectors.status(state),
  // latestBlock: nodeSelectors.latestBlock(state),
  // currentBlock: nodeSelectors.currentBlock(state)
})

export const mapDispatchToProps = dispatch =>
  bindActionCreators(
    {
      // getStatus: nodeHandlers.epics.getStatus
    },
    dispatch
  )

export default connect(mapStateToProps, mapDispatchToProps)(SyncLoader)
