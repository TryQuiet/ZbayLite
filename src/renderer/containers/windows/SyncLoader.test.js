import { mapStateToProps } from './SyncLoader'
import create from '../../store/create'
import { NodeState } from '../../store/handlers/node'

jest.mock('../../../shared/electronStore', () => ({
  set: () => {},
  get: () => {}
}))

describe('SyncLoader', () => {
  let store = null
  beforeEach(() => {
    jest.clearAllMocks()
    store = create({
      initialState: {
        node: {
          ...NodeState,
          currentBlock: 1,
          latestBlock: 100
        }
      }
    })
  })

  it('will receive right props', async () => {
    const props = mapStateToProps(store.getState())
    expect(props).toMatchSnapshot()
  })
})
