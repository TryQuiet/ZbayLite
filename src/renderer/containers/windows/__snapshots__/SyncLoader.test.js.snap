// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SyncLoader will receive right props 1`] = `
Object {
  "currentBlock": 1,
  "latestBlock": 100,
  "nodeStatus": "healthy",
}
`;
